File: UI/src/lib/types/ai-settings.ts
export interface AISettings {
	provider: string;
	base_url: string;
	model: string;
	has_api_key: boolean;
	description_expand_prompt: string;
	default_prompt: string;
	issue_copy_prompt: string;
	default_issue_copy_prompt: string;
	created_at: string;
	updated_at: string;
}

export interface UpdateAISettingsRequest {
	provider?: string;
	base_url?: string;
	model?: string;
	api_key?: string | null;
	description_expand_prompt?: string;
	issue_copy_prompt?: string;
}

File: BE/internal/domain/ai_settings.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

const AIProviderOpenAICompatible = "openai_compatible"

type AISettings struct {
	WorkspaceID             uuid.UUID `json:"workspace_id" db:"workspace_id"`
	Provider                string    `json:"provider" db:"provider"`
	BaseURL                 string    `json:"base_url" db:"base_url"`
	Model                   string    `json:"model" db:"model"`
	APIKeyEncrypted         *string   `json:"-" db:"api_key_encrypted"`
	DescriptionExpandPrompt string    `json:"description_expand_prompt" db:"description_expand_prompt"`
	CreatedAt               time.Time `json:"created_at" db:"created_at"`
	UpdatedAt               time.Time `json:"updated_at" db:"updated_at"`
}
